src/main/java/org/broadinstitute/hellbender/engine/dataflow/datasources/VariantsDataflowSource.java

```diff
@@ -0,0 +1,88 @@
+package org.broadinstitute.hellbender.engine.dataflow.datasources;
+
+import com.google.cloud.dataflow.sdk.Pipeline;
+import com.google.cloud.dataflow.sdk.transforms.Create;
+import com.google.cloud.dataflow.sdk.values.PCollection;
+import htsjdk.tribble.Feature;
+import htsjdk.tribble.FeatureCodec;
+import htsjdk.variant.variantcontext.VariantContext;
+import org.broadinstitute.hellbender.engine.FeatureDataSource;
+import org.broadinstitute.hellbender.engine.FeatureManager;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.utils.dataflow.BucketUtils;
+import org.broadinstitute.hellbender.utils.variant.Variant;
+import org.broadinstitute.hellbender.utils.variant.VariantContextVariantAdapter;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ * VariantsDataflowSource loads variants into PCollections using a local file (GCS Bucket to come).
+ *
+ */
+public class VariantsDataflowSource {
+
+    private final List<String> variantSources;
+    private final Pipeline pipeline;
+
+    /**
+     * VariantsDataflowSource sets up source using files (or eventually GCS buckets).
+     * @param variantSources, list of files (or eventually buckets) to read from
+     * @param pipeline, options to get credentials to access GCS buckets.
+     */
+    public VariantsDataflowSource(final List<String> variantSources, final Pipeline pipeline) {
```

Comment:
I'd prefer the argument be named "variantFile" or "variantUrl" - it's confusing to have"Source"  being used on the same line to mean other "will return a PCollection" or "File or URL".